main.py

```diff
@@ -90,47 +96,56 @@ async def add_preference(preference: UserPreference):
 async def get_preferences():
     return {"preferences": user_preferences}
 
+
 @app.post("/tourguide")
 async def query_everywhere_tourguide(
-        request: TourGuideRequest, location: dict[str, str] = Depends(get_location)
+    request: TourGuideRequest, location: dict[str, str] = Depends(get_location)
 ):
     if not request.location:
-        request.location = reverse_geocode(ReverseGeocode(lat=request.lat, lng=request.lon))
+        request.location = reverse_geocode(
+            ReverseGeocode(lat=request.lat, lng=request.lon)
+        )
 
+    # Ensure images are base64 strings without the data URL prefix
     request.base_image = request.base_image.replace("data:image/jpeg;base64,", "")
     request.masked_image = request.masked_image.replace("data:image/jpeg;base64,", "")
 
-    if request.stream:   
+    if request.stream:
         return StreamingResponse(
-                run_everywhere_tour_guide(
-                    base_image=request.base_image,
-                    masked_image=request.masked_image,
-                    location=request.location,
-                    lat=request.lat,
-                    lon=request.lon
-                ),
-                media_type="text/plain"
+            run_everywhere_tour_guide(
+                base_image=request.base_image,
+                masked_image=request.masked_image,
+                location=request.location,
+                lat=request.lat,
+                lon=request.lon,
+            ),
+            media_type="text/plain",
         )
 
-    response = "".join([chunk async for chunk in run_everywhere_tour_guide(
-        base_image=request.base_image,
-        masked_image=request.masked_image,
-        location=request.location,
-        lat=request.lat,
-        lon=request.lon)])
+    response = "".join(
+        [
+            chunk
+            async for chunk in run_everywhere_tour_guide(
+                base_image=request.base_image,
+                masked_image=request.masked_image,
+                location=request.location,
+                lat=request.lat,
+                lon=request.lon,
+            )
+        ]
+    )
     return response
 
 
 @app.post("/query_assistant")
 async def query_assistant(
     query: Query, preferences: dict[str, list[str]] = Depends(get_preferences)
 ):
-
     previous_conversation = conversations.get(query.id, [])
 
     assistant = Assistant(
-        client=AsyncAnthropic(api_key=os.getenv("ANTHROPIC_API_KEY")),
-        model="claude-3-5-haiku-latest",
+        client=client,
```

Comment:
Have you updated how the messages get reset? We ran into an issue with this where if the FastAPI endpoint gets multiple requests it just hits the `max_messages` (even if it's the users first message, because it's from other users messages).

Reply:
Right, updated the conversation handler, which uses in-session memory. Let me know if the logic is good